### PR TITLE
Create post

### PR DESCRIPTION
**Pull request by NgengChhengkim**

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class PostsController < ApplicationController
+  def create
```
Comment (yongsokheng) [inline app/controllers/posts_controller.rb:4]: load cancancan

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    @post = current_user.posts.new post_params
+    if @post.save
+      current_user.activities.create trackable_id: @post.id, trackable_type: @post.class.name
```
Comment (yongsokheng) [inline app/controllers/posts_controller.rb]: @post.create_activity user_id: current_user.id

File: app/controllers/static_pages_controller.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 class StaticPagesController < ApplicationController
   def home
     @activities = Activity.all
+    @job_types = JobType.all.limit(10)
```
Comment (yongsokheng) [inline app/controllers/static_pages_controller.rb]: Settings

File: app/models/job_post.rb [hunk 4]
```diff
@@ -4,10 +4,29 @@ class JobPost < ApplicationRecord
...
+
   delegate :name, to: :job_type, prefix: true, allow_nil: true
   delegate :user_id, :content, to: :post, prefix: true, allow_nil: true
 
   scope :by_user, ->user_id do
     joins(:post).where("posts.user_id = ?", user_id).order created_at: :desc
   end
+
+  private
+  def job_post_error
```
Comment (yongsokheng) [inline app/models/job_post.rb]: dung lam gi?

File: app/models/job_post.rb [hunk 5]
```diff
@@ -4,10 +4,29 @@ class JobPost < ApplicationRecord
...
+    if errors.any?
+      errors.clear
+      post.errors.clear
+      post.errors.add :base, I18n.t("errors.messages.job_post_error")
+    end
+  end
+
+  def job_content
+     errors.add(:base, "content blank") if post.content.blank?
```
Comment (yongsokheng) [inline app/models/job_post.rb]: dung lam gi?

File: app/models/post.rb [hunk 6]
```diff
@@ -1,15 +1,29 @@
...
 
-  has_many :activities, as: :trackable, dependent: :destroy
+  has_one :activity, as: :trackable, dependent: :destroy
   has_many :share_posts, dependent: :destroy
 
+  validate :validate_post
+
   enum status: [:public_post, :private_post]
 
+  accepts_nested_attributes_for :job_post, allow_destroy: true
```
Comment (yongsokheng) [inline app/models/post.rb]: allow_destroy?

File: app/models/share_post.rb [hunk 7]
```diff
@@ -2,7 +2,7 @@ class SharePost < ApplicationRecord
   belongs_to :user
   belongs_to :post
 
-  has_many :activities, as: :trackable, dependent: :destroy
+  has_one :activities, as: :trackable, dependent: :destroy
```
Comment (yongsokheng) [inline app/models/share_post.rb]: has_one :activity, as: :trackable, dependent: :destroy

File: app/controllers/static_pages_controller.rb [hunk 8]
```diff
@@ -1,5 +1,6 @@
 class StaticPagesController < ApplicationController
   def home
     @activities = Activity.all
+    @job_types = JobType.all
```
Comment (NgengChhengkim) [inline app/controllers/static_pages_controller.rb:4]: se lam trong pull khac. pull selectize data

Comment (yongsokheng) [reply app/controllers/static_pages_controller.rb:4]: OK

File: app/views/posts/_job_post_form.html.erb [hunk 9]
```diff
@@ -0,0 +1,36 @@
+<%= form_for Post.new, url: posts_path, method: :post, remote: true,
```
Comment (yongsokheng) [inline app/views/posts/_job_post_form.html.erb]: Post.new de trong controller 
bo method post
bo url
`form_for @post, remote: true` la du roi

File: app/views/posts/_job_post_form.html.erb [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+  html: {id: "new-job-post"} do |f| %>
+  <div id="job-post-errors"></div>
+
+  <%= f.fields_for :job_post, f.object.build_job_post do |builder| %>
```
Comment (yongsokheng) [inline app/views/posts/_job_post_form.html.erb]: build_job_post viet knong controller . knong neng ot cham bach viet ey te

```
@post= Post.new
@post.build_job_post
```

File: app/views/posts/_job_post_form.html.erb [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    </div><br />
+    <div>
+      <%= builder.text_field :location, placeholder: t("posts.location"), class: "job-location form-control" %>
+    </div><br />
+    <div>
+      <%= builder.select :job_type_id, options_for_select(JobType.all.map{|u| [u.name, u.id]}),
+      { prompt: t("posts.job_type")}, {id: "job-type-post"} %>
+    </div><br />
+
+    <div><%= f.text_area :content, id: "post-content" %></div>
```
Comment (yongsokheng) [inline app/views/posts/_job_post_form.html.erb:16]: min bach dak height knong css te, sak brer rows cheng merl der ot
`f.text_area :content, rows: 2`

File: app/views/posts/_job_post_form.html.erb [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+  <% end %>
+
+  <div class="button">
+    <label class="btn btn-default btn-file pull-left">
+      <%= t "posts.upload_photo" %>
+      <input type="file" style="display: none;" name="post[image]" id="post_image">
```
Comment (yongsokheng) [inline app/views/posts/_job_post_form.html.erb]: `f.object.file :post_image`

File: app/views/posts/_post_form.html.erb [hunk 13]
```diff
@@ -0,0 +1,20 @@
+<%= form_for :post, url: posts_path, method: :post, remote: true, html: {id: "new-post"} do |f| %>
```
Comment (yongsokheng) [inline app/views/posts/_post_form.html.erb]: nhu tren

File: app/views/posts/_post_form.html.erb [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+  <div id="post-errors"></div>
+  <div><%= f.text_area :content, placeholder: t("posts.write_here") ,class: "form-control" %></div>
+
+  <div class="button">
+    <label class="btn btn-default btn-file pull-left">
+      <%= t "posts.upload_photo" %>
+      <input type="file" style="display: none;" name="post[image]" id="post_image">
```
Comment (yongsokheng) [inline app/views/posts/_post_form.html.erb]: nhu tren

File: app/views/posts/_post_form.html.erb [hunk 15]
```diff
@@ -0,0 +1,20 @@
+<%= form_for :post, url: posts_path, method: :post, remote: true, html: {id: "new-post"} do |f| %>
+  <div id="post-errors"></div>
+  <div><%= f.text_area :content, placeholder: t("posts.write_here") ,class: "form-control" %></div>
+
+  <div class="button">
```
Comment (yongsokheng) [inline app/views/posts/_post_form.html.erb]: doch khnea, brer partial tov

File: app/views/posts/_job_post_form.html.erb (hunk 11 above)
Comment (NgengChhengkim) [reply app/views/posts/_job_post_form.html.erb:16]: ort der te

File: app/models/education.rb [hunk 16]
```diff
@@ -9,5 +9,7 @@ class Education < ApplicationRecord
   validates :start_date, presence: true
   validates :end_date, presence: true
 
+  validates_uniqueness_of :degree_id, scope: :user_id
```
Comment (yongsokheng) [inline app/models/education.rb]: pull neng mech tak tong with education.rb ?

File: app/views/posts/_button.erb [hunk 17]
```diff
@@ -0,0 +1,15 @@
+<div class="button">
```
Comment (yongsokheng) [inline app/views/posts/_button.erb]: app/views/posts/_button.html.erb

File: app/views/posts/_button.erb [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+  <label class="btn btn-default btn-file pull-left">
+    <%= t "posts.upload_photo" %>
+    <input type="file" style="display: none;" name="post[image]" id="post_image">
```
Comment (yongsokheng) [inline app/views/posts/_button.erb]: brer f.file :image ot ket te?

File: app/views/posts/_job_post_form.html.erb (hunk 11 above)
Comment (yongsokheng) [reply app/views/posts/_job_post_form.html.erb:16]: f.text_area :content, size: "30x10"

File: app/views/posts/create.js.erb [hunk 19]
```diff
@@ -0,0 +1,18 @@
+<% if @post.errors.any? %>
+  <% if @post.job_post.nil? %>
+    $("#post-errors").html("<div class='text-danger'><%= t 'posts.post_error' %></div>");
```
Comment (yongsokheng) [inline app/views/posts/create.js.erb:3]:
```
error_element = $("#post-errors");
else
error_element = $("#job-post-errors")
...
error_element.html("<div class='text-danger'><%= t 'posts.job_post_error' %></div>");
```

File: app/views/posts/_job_post_form.html.erb (hunk 11 above)
Comment (NgengChhengkim) [reply app/views/posts/_job_post_form.html.erb:16]: mech ban ort dak css.

Comment (NgengChhengkim) [reply app/views/posts/_job_post_form.html.erb:16]: ?

File: app/views/posts/_button.html.erb [hunk 20]
```diff
@@ -0,0 +1,15 @@
+<div class="button">
+  <label class="btn btn-default btn-file pull-left">
+    <%= t "posts.upload_photo" %>
+    <input type="file" style="display: none;" name="post[image]" id="post_image">
```
Comment (yongsokheng) [inline app/views/posts/_button.html.erb:4]: brer input file bos rail ot ket te?

File: app/views/posts/_job_post_form.html.erb (hunk 11 above)
Comment (yongsokheng) [reply app/views/posts/_job_post_form.html.erb:16]: dak cheng sruol merl cheang :smile:

File: app/views/posts/_button.html.erb (hunk 20 above)
Comment (NgengChhengkim) [reply app/views/posts/_button.html.erb:4]: brer file_field  dak class form-control chol ort skal

File: app/views/posts/_job_post_form.html.erb (hunk 11 above)
Comment (NgengChhengkim) [reply app/views/posts/_job_post_form.html.erb:16]: cheng ey pseng teat dak neng tang os mong. kchil brer css

File: app/views/posts/create.js.erb (hunk 19 above)
Comment (NgengChhengkim) [reply app/views/posts/create.js.erb:3]: error pseng knea

File: app/views/posts/_button.html.erb (hunk 20 above)
Comment (yongsokheng) [reply app/views/posts/_button.html.erb:4]: display: none neng, dak form-control tver ey?

File: app/views/posts/_job_post_form.html.erb (hunk 11 above)
Comment (yongsokheng) [reply app/views/posts/_job_post_form.html.erb:16]: OK

**Review by yongsokheng — CHANGES_REQUESTED** [review]

File: app/views/posts/create.js.erb (hunk 19 above)
Comment (yongsokheng) [reply app/views/posts/create.js.erb:3]: OK

File: app/views/posts/_button.html.erb (hunk 20 above)
Comment (NgengChhengkim) [reply app/views/posts/_button.html.erb:4]: btn btn-default

File: app/models/education.rb (hunk 16 above)
Comment (yongsokheng) [reply app/models/education.rb]: not yet reply ??

File: app/views/posts/_button.html.erb (hunk 20 above)
Comment (yongsokheng) [reply app/views/posts/_button.html.erb:4]: ok

**Review by yongsokheng — APPROVED** [review]

**Pull request merged**